gunicorn/http/body.py

```diff
@@ -257,3 +257,12 @@ def readlines(self, size=None):
                 line, data = data[:pos + 1], data[pos + 1:]
                 ret.append(line)
         return ret
+
+    def discard(self):
+        discarded_len = 0
+        bytes_read = six.MAXSIZE
+        while bytes_read > 0:
```

Comment:
`self.reader` here is already terminated by the HTTP framing (content-length or chunks) right? If so this seems fine (otherwise it's a new DOS opportunity).

Reply:
I would just iterate the reader there, since it's terminated. Like the `req.body.readlines()` without appending the data.